### PR TITLE
Add catch2 v3.3.0

### PR DESCRIPTION
**Pull request by sebastianbergt**
Simply add the newest catch2 version available right now.

### CODE REVIEW

File: modules/catch2/metadata.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
         "github:catchorg/Catch2"
     ],
     "versions": [
```
Comment (fmeum) [inline modules/catch2/metadata.json:13]: Please keep this sorted.

File: modules/catch2/3.3.0/source.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "url": "https://github.com/catchorg/Catch2/archive/refs/tags/v3.3.0.tar.gz",
+    "integrity": "sha256-/i8ppUyndcLdBLuX/7edOY5iEOPKoXQ0i1zTt+TKiH0=",
+    "strip_prefix": "Catch2-3.3.0",
+    "patch_strip": 0,
+    "patches": {
+        "module_dot_bazel.patch": "sha256-0ZKA9VjsNb7SM6kYVsfPs+XBKKFVGTYgnHFpohIOkXA="
```
Comment (fmeum) [inline modules/catch2/3.3.0/source.json:7]: This integrity value has to be updated as the module file patch has changed.

Comment (sebastianbergt) [inline modules/catch2/3.3.0/source.json:7]:
```suggestion
        "module_dot_bazel.patch": "sha256-Ssp5WC3aSXutLRpglAFDmlxvtV3B9SDP1rKpI0tAfvE="
```

File: modules/catch2/metadata.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
...
+        "3.3.0",
         "3.2.1"
```
Comment (sebastianbergt) [inline modules/catch2/metadata.json:15]:
```suggestion
        "3.2.1",
        "3.3.0"
```

File: modules/catch2/metadata.json (hunk 1 above)
Comment (sebastianbergt) [reply modules/catch2/metadata.json:13]: Thank you for the quick review.
I guess you are referring to an ascending order.
Changed it accordingly

File: modules/catch2/3.3.0/source.json (hunk 2 above)
Comment (sebastianbergt) [reply modules/catch2/3.3.0/source.json]: Updated it. Thank you for pointing it out.

**Review by fmeum — APPROVED** [review]
Thanks!

**Pull request merged**